metaflow/plugins/aws/batch/batch_client.py

```diff
@@ -575,6 +582,28 @@ def __init__(self, group, stream, pos=0, sleep_on_no_data=0):
         self._token = None
 
     def _get_events(self):
+        done = RunningJob.is_done
+        status_code = RunningJob.status_code
+        if done:
```

Comment:
This should be done later probably as there may still be messages being pulled below. I suggest checking this at the end (before the return) and appending the message to that array.

Reply:
I moved the variables and if statement below the try and except block under `_get_events(self)`. 

- [x] Fixed